R/cut.R

```diff
@@ -0,0 +1,132 @@
+                                             include.lowest = FALSE,
+                                             right = TRUE,
+                                             dig.lab = 3,
+                                             ordered_result = FALSE, ...){
+    env <- environment()
+      if (missing(name)) {
+        halt("Must provide the name for the new variable")
+    }
+    if (length(breaks) == 1L) {
+        if (is.na(breaks) || breaks < 2L) {
+            halt("invalid number of intervals")
+        }
+        nb <- as.integer(breaks + 1) # one more than #{intervals}
+        rx <- c(min(x, na.rm = TRUE), max(x, na.rm = TRUE))
+        dx <- diff(rx)
+        if (dx == 0) {
+            dx <- abs(rx[1L])
+            breaks <- seq.int(rx[1L] - dx/1000, rx[2L] + dx/1000,
+                              length.out = nb)
+        } else {
+            breaks <- seq.int(rx[1L], rx[2L], length.out = nb)
+            breaks[c(1L, nb)] <- c(rx[1L] - dx/1000, rx[2L] + dx/1000)
+        }
+    }else {
+        breaks <- sort.int(as.double(breaks))
+        nb <- length(breaks)
+    }
+    if (anyDuplicated(breaks)) halt("'breaks' are not unique")
+    if (is.null(labels)) { #Autogenerate labels if not supplied
+      labels <- generateCutLabels(dig.lab, breaks, nb, right, include.lowest)
+    } else if (length(labels) != nb - 1L) {
+        stop("lengths of 'breaks' and 'labels' differ")
+    } 
+    if (right) {
+        `%c1%` <- function(x,y) x <= y
+        `%c2%` <- function(x,z) x > z
+    } else {
+        `%c1%` <- function(x,y) x < y
+        `%c2%` <- function(x,z) x >= z
+    }
+
+    cases <- vector("list", length = length(breaks) - 1)
+    
+    for (i in 2:length(breaks)) {
+        cases[[i - 1]] <- x %c2% breaks[i - 1] & x %c1% breaks[i]
+    }
+    case_list <- lapply(seq_along(cases), function(x) list(expression = cases[[x]], name = labels[x]))
+    makeCaseVariable(cases = case_list, name = name, ...)
+}
+)
+
+
+#' Generate Labels for the cut function
+#' 
+#' A convenience function to generate labels for the cut function. This
+#' function is extracted from [base::cut()] and is broken out to make it easier to 
+#' test. It is not meant to be called on its own. 
+#'
+#' @param dig.lab see `cut()`
+#' @param breaks see `cut()`
+#' @param nb The number of breaks, equal to the small of 2 or the number of breaks
+#' @param right  see `cut()`
+#' @param include.lowest see`cut()`
+#'
+#' @return
+#' A character vector of labels
+#' @examples
+#' 
+#' crunch:::generateCutLabels(2, c(2, 3, 4, 5), 4, FALSE, FALSE)
+#' 
+generateCutLabels <- function(dig.lab, breaks, nb, right, include.lowest) {
+    for (dig in dig.lab:max(12L, dig.lab)) {
+        ## 0+ avoids printing signed zeros as "-0"
+        ch.br <- formatC(0 + breaks, digits = dig, width = 1L)
+        ok <- all(ch.br[-1L] != ch.br[-nb])
+        if (ok ) break
+    }
+    labels <- if (ok) {
+            paste0(if (right) "(" else "[",
```

Comment:
I think `ifelse` would be cleaner here since everything is all on one line. The guidelines I've been using is: 
```
if (foo) {
    bar
} else {
    baz
}
```
Or if it is that simple and I want it on one line `ifelse(foo, bar, baz)`